lib/src/matching_strategy_enum.dart

```diff
@@ -0,0 +1,18 @@
+  enum MatchingStrategy { 
```

Comment:
Unfortunately, I will have to create a breaking change if I accept this PR. 

This requires the 'enhanced-enums' language feature to be enabled, which requires a minimum SDK version of 2.17 or higher, and currently, we support 2.12 (with users using it).

So I think we will have to stick to old-school strings 😢

Reply:
Oh, are you positive that's the case? I'm asking because I was able to run the tests with the provided docker image without any errors using this code. I only got an `'enhanced-enums' 2.17 version error` when trying to define the `name` method using a newer approach, which doesn't work. And that's I had to use `extension`. 

I see the integration tests now are complaining about that `semicolon` I added when trying the new approach. I'll replace it with a comma. Let's see.

Reply:
My vscode is dying I think that's the reason haha, thanks. Indeed it is working normally. There is just a small issue regarding the lining, but after that we're good to go!